### ISSUE TITLE
Congruity and Firefox

### ISSUE THREAD

**Issue by foberle** [jaymzh/concordance#12]
Hi:

I've used the Concordance and Congruity apps for several years to reconfigure my Logitech Harmony One remote control on the Members.HarmonyRemote.com site, but I'm unable to log in to their new site (https://setup.myharmony.com/).

I believe that the problem is that I haven't configured Firefox to report its capabilities properly (although it seems straightforward enough, and I've done it for other sites).

I believe I have the most updated versions of Concordance, Congruity, and Firefox, and am running on a regularly updated version of Ubuntu 16.04.

Can you tell me how exactly I can use Logitech's newest site (or even if its possible) as I added another remote (650) that can't be updated with the "legacy" site?

Thanks in advance.

**Comment by swt2c** [comment]
Unfortunately setup.myharmony.com uses Silverlight and thus you can't just trick Firefox to load it.

For the newer site, you will have to use mhgui which is essentially a front end around the setup.myharmony.com backend.  It is distributed with congruity.  You'll probably want to use the version from git and not the one that comes with Ubuntu as it will unfortunately be outdated.

**Comment by foberle** [comment]
Hi Scott:

Thanks for the response, but I suspect I'm in a sort of cloud here. I just downloaded a file:
congruity-18.tar.bz2 - But: the date on the archive is 29 March 2014, so I'm obviously in the wrong place or I misinterpreted your response.

Also, I notice that Ubuntu has a manpage for mhgui (which references your name) and has a link to the file mhgui.1.gz (dated 18 April of this year), which I also downloaded, but haven't attempted to extract. Is this the "version from git" you referred to?

Do I simply need to install the latter, or were you suggesting that I need to load some new version of congruity as well?

Can I have the old and new elements installed simultaneously? My Harmony One works just fine with the "original" members site, but the new 650 requires the Silverlight page. And, do you have a suggested recommendation for the OS alias I need to use with Firefox?

OR: Do you know of anyone who has written up all of this so I can quit bothering you (if you are the developer who set all this up, we all owe you a debt of gratitude for sharing it)?

Thanks for responding regardless...

**Comment by foberle** [comment]
Hi again Scott:

Of course, as soon as I sent the last message, I stumbled on to your page:
https://www.mail-archive.com/concordance-devel@lists.sourceforge.net/msg01490.html

I confirmed that I do have the python-suds version 0.7 installed on my system (although I have no idea why or what it's currently used for, although the gramps genealogy program is written in python).

So, for the heck of it, I extracted mhgui and ran ./mhgui.1, which gave me the following:
./mhgui.1: line 1: .TH: command not found
./mhgui.1: line 2: .SH: command not found
^CTraceback (most recent call last):
  File "/usr/bin/mhgui", line 50, in <module>
    mhMgr = MHManager(use_local_wsdl, suds_debug)
  File "/usr/share/congruity/mhmanager.py", line 156, in __init__
    self.client = Client(url, cache=cache, plugins=[MHPlugin()])
  File "/usr/lib/python2.7/dist-packages/suds/client.py", line 120, in __init__
    self.wsdl = reader.open(url)
  File "/usr/lib/python2.7/dist-packages/suds/reader.py", line 104, in open
    wsdl = self.fn(url, self.options)
  File "/usr/lib/python2.7/dist-packages/suds/wsdl.py", line 188, in __init__
    self.open_imports(imported_definitions)
  File "/usr/lib/python2.7/dist-packages/suds/wsdl.py", line 233, in open_imports
    imp.load(self, imported_definitions)
  File "/usr/lib/python2.7/dist-packages/suds/wsdl.py", line 354, in load
    d = Definitions(url, definitions.options, imported_definitions)
  File "/usr/lib/python2.7/dist-packages/suds/wsdl.py", line 164, in __init__
    d = reader.open(url)
  File "/usr/lib/python2.7/dist-packages/suds/reader.py", line 148, in open
    xml = self.__fetch(url)
  File "/usr/lib/python2.7/dist-packages/suds/reader.py", line 188, in __fetch
    fp = self.options.transport.open(request)
  File "/usr/lib/python2.7/dist-packages/suds/transport/https.py", line 62, in open
    return HttpTransport.open(self, request)
  File "/usr/lib/python2.7/dist-packages/suds/transport/http.py", line 66, in open
    return self.u2open(u2request)
  File "/usr/lib/python2.7/dist-packages/suds/transport/http.py", line 127, in u2open
    return url.open(u2request, timeout=tm)
  File "/usr/lib/python2.7/urllib2.py", line 429, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 447, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 1228, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.7/urllib2.py", line 1195, in do_open
    h.request(req.get_method(), req.get_selector(), req.data, headers)
  File "/usr/lib/python2.7/httplib.py", line 1057, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1097, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 897, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 859, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 836, in connect
    self.timeout, self.source_address)
  File "/usr/lib/python2.7/socket.py", line 566, in create_connection
    sock.connect(sa)
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
KeyboardInterrupt
./mhgui.1: line 4: .SH: command not found
./mhgui.1: line 5: .B: command not found
./mhgui.1: line 6: [OPTIONS]: command not found
./mhgui.1: line 7: .SH: command not found
./mhgui.1: line 8: .B: command not found
./mhgui.1: line 9: is: command not found
./mhgui.1: line 10: enable: programming: not a shell builtin
./mhgui.1: line 10: enable: of: not a shell builtin
./mhgui.1: line 10: enable: Logitech: not a shell builtin
./mhgui.1: line 10: enable: Harmony: not a shell builtin
./mhgui.1: line 10: enable: remote: not a shell builtin
./mhgui.1: line 10: enable: controls: not a shell builtin
./mhgui.1: line 10: enable: that: not a shell builtin
./mhgui.1: line 10: enable: are: not a shell builtin
./mhgui.1: line 10: enable: only: not a shell builtin
./mhgui.1: line 10: enable: supported: not a shell builtin
./mhgui.1: line 11: syntax error near unexpected token `('
./mhgui.1: line 11: `through this web site (which normally requires the use of Microsoft'
... at which point, the world stopped, and I used Ctrl+C to return.

Whether or not this failure is because I wasn't on-line with Harmony isn't clear to me, since the last programming I did predates Python as far as I know (yes, I'm an old retired fart).

Again, thanks for reading my posts - if you can walk me through all this, I'd be very grateful.

Frank

**Comment by swt2c** [comment]
Sorry this isn't documented better.  :-)  Wow, you found my original email for mhgui.  Things have changed a bit since then.

mhgui.1 is the man page, so you can't run that.

To get the latest version from git and run it (it will go into the current directory):

```
git clone git://git.code.sf.net/p/congruity/code congruity
cd congruity
./mhgui
```

**Comment by foberle** [comment]
Thanks very much Scott.

Right after sending the last message, I decided to read through the code and only then realized it was a man page. I've been away from the world of IT for too long it seems.

Once "git-ting" the actual mhgui, things worked quite a bit better. I very much like the idea of having a dedicated app that chats with the server rather than the "old" way of having the process flip back and forth. I guess I just incorrectly assumed that the new version worked exactly the same as the previous system which is why I was concerned about Firefox: so kudos for the direct attack.

Of course, it's annoying that Logitech wants us to use one method for the Harmony One and one for the 650. For what are (in my view anyway) far and away the best handheld remotes, their creator's approach is rather non user-friendly. Now that I feel safe in acquiring a 650, I'll do so and contact you if I have further questions.

Thanks again for your prompt assistance.

**Comment by foberle** [comment]
Hi Scott: 

I hate to bother you again, but I've run into a problem while attempting to make new changes to my old Harmony One, and there seems to be no support from Logitech for use with Linux (they were fairly abrupt, actually) and I was wondering if you had any suggestions. Basically I'm unable to program my Harmony One any longer using the members.harmonyremote.com site. 

The only change to my system I can think of was that I used your new mhgui to set up another account (for planned purchase of another Logitech model not supported by members.harmonyremote), so I'm wondering if there is some connection to that. 

The symptom is fairly straightforward: When I get to the screen that says: 
============================================================================================================= 
Click next to: 
ensure the Harmony One can communicate properly with the remote software 
detect the firmware version 
============================================================================================================= 
the EZHex file that had previously been executed automatically to confirm the connection pops up in a text editor instead. 

I located the file in /tmp/mozilla_foberle0/Connectivity.EZHex; it showed that I was the owner, but that there was only read permission. 

I attempted to add executable permission (chmod +x), and run it manually, but got the following errors: 
============================================================================================================= 
./Connectivity.EZHex: line 1: ?xml: No such file or directory 
./Connectivity.EZHex: line 2: INFORMATION: No such file or directory 
./Connectivity.EZHex: line 3: USERMESSAGES: No such file or directory 
./Connectivity.EZHex: line 4: syntax error near unexpected token `<' 
'/Connectivity.EZHex: line 4: ` <USERMESSAGE><VERSIONS> 
============================================================================================================= 
I suspect that all this proves is that I don't know what I'm doing, but just in case. 

The contents of the file are: 
============================================================================================================= 
<?xml version="1.0"?> << THIS IS LINE 1 
<INFORMATION> << THIS IS LINE 2 
<USERMESSAGES> << THIS IS LINE 3 
<USERMESSAGE><VERSIONS> << THIS IS LINE 4 
<VERSION> 
<CLIENTSOFTWARE>2.3</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.4</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.5</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.7</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.8</CLIENTSOFTWARE> 
</VERSION> 
</VERSIONS> 
<TYPE>Warning</TYPE> <TEXT>WARNING: This version of the software is no longer supported. Please click on the 'Software' link in the top right corner of your browser to get the latest version.</TEXT> </USERMESSAGE> <USERMESSAGE><VERSIONS> 
<VERSION> 
<CLIENTSOFTWARE>2.3</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.4</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.5</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.7</CLIENTSOFTWARE> 
</VERSION> 
<VERSION> 
<CLIENTSOFTWARE>2.8</CLIENTSOFTWARE> 
</VERSION> 
</VERSIONS> 
<TYPE>Abort</TYPE> </USERMESSAGE> </USERMESSAGES> 
<TRANSFERTYPE>Winsock</TRANSFERTYPE> <CONFIGURATION> 
<CONFIGOPTION> 
<KEY>InstructionNoHarmony</KEY> 
<TITLE>No Remote Detected</TITLE> 
<IMAGE>0</IMAGE> 
<VALUE>Plug the USB cable to the remote, or press any button on the remote.</VALUE> 
</CONFIGOPTION> 
<CONFIGOPTION> 
<KEY>GetZapsOnly</KEY> 
<VALUE>True</VALUE> 
</CONFIGOPTION> 
</CONFIGURATION> 
<POSTOPTIONS> 
<SERVER>members.harmonyremote.com</SERVER> 
<PATH>EasyZapper/New/ProcConnectivity/Connectivity_Receive_Zaps.asp</PATH> 
<TIMEOUT>60000</TIMEOUT> <HEADERS> <HEADER> <KEY>Cookie</KEY> <VALUE>HR%5FUSER%5FLANGUAGE=enu; SKIN%5FID=Harmony; PSWD=oo1S8zO%2FVTo%2Fo3qIrvD5aQ%3D%3D; AUSER=foberle; CookieKeyValue=%7BEEEEEEEE%2DEEEE%2DEEEE%2DEEEE%2DEEEEEEEEEEEE%7D%7B5D13C4F1%2DA652%2D4C25%2D9B19%2DABE89CEC0B17%7D%7B494C76E0%2D1DDD%2D40C2%2D9DDE%2D1ACAA64CE509%7D; ASPSESSIONIDQASATDRA=OIOHLMMCCGNJMJPOMONFBFEG; HRLB=172257797751533668</VALUE> </HEADER> </HEADERS> <PARAMETERS> 
<PARAMETER> 
<KEY>UserId</KEY> 
<VALUE>8014371</VALUE> 
</PARAMETER> 
</PARAMETERS> 
</POSTOPTIONS> 
<TIPPOSTOPTIONS> 
<SERVER>members.harmonyremote.com</SERVER> 
<TIMEOUT>60000</TIMEOUT> <HEADERS> <HEADER> <KEY>Cookie</KEY> <VALUE>HR%5FUSER%5FLANGUAGE=enu; SKIN%5FID=Harmony; PSWD=oo1S8zO%2FVTo%2Fo3qIrvD5aQ%3D%3D; AUSER=foberle; CookieKeyValue=%7BEEEEEEEE%2DEEEE%2DEEEE%2DEEEE%2DEEEEEEEEEEEE%7D%7B5D13C4F1%2DA652%2D4C25%2D9B19%2DABE89CEC0B17%7D%7B494C76E0%2D1DDD%2D40C2%2D9DDE%2D1ACAA64CE509%7D; ASPSESSIONIDQASATDRA=OIOHLMMCCGNJMJPOMONFBFEG; HRLB=172257797751533668</VALUE> </HEADER> </HEADERS> </TIPPOSTOPTIONS> 
<TIMEPOSTOPTIONS> 
<SERVER>members.harmonyremote.com</SERVER> 
<PATH>EasyZapper/GetTime.asp</PATH> 
<TIMEOUT>60000</TIMEOUT> <HEADERS> <HEADER> <KEY>Cookie</KEY> <VALUE>HR%5FUSER%5FLANGUAGE=enu; SKIN%5FID=Harmony; PSWD=oo1S8zO%2FVTo%2Fo3qIrvD5aQ%3D%3D; AUSER=foberle; CookieKeyValue=%7BEEEEEEEE%2DEEEE%2DEEEE%2DEEEE%2DEEEEEEEEEEEE%7D%7B5D13C4F1%2DA652%2D4C25%2D9B19%2DABE89CEC0B17%7D%7B494C76E0%2D1DDD%2D40C2%2D9DDE%2D1ACAA64CE509%7D; ASPSESSIONIDQASATDRA=OIOHLMMCCGNJMJPOMONFBFEG; HRLB=172257797751533668</VALUE> </HEADER> </HEADERS> </TIMEPOSTOPTIONS> 
<COMPLETEPOSTOPTIONS> 
<SERVER>members.harmonyremote.com</SERVER> 
<PATH>EasyZapper/New/ProcConnectivity/Connectivity_Receive_Complete.asp</PATH> 
<TIMEOUT>60000</TIMEOUT> <HEADERS> <HEADER> <KEY>Cookie</KEY> <VALUE>HR%5FUSER%5FLANGUAGE=enu; SKIN%5FID=Harmony; PSWD=oo1S8zO%2FVTo%2Fo3qIrvD5aQ%3D%3D; AUSER=foberle; CookieKeyValue=%7BEEEEEEEE%2DEEEE%2DEEEE%2DEEEE%2DEEEEEEEEEEEE%7D%7B5D13C4F1%2DA652%2D4C25%2D9B19%2DABE89CEC0B17%7D%7B494C76E0%2D1DDD%2D40C2%2D9DDE%2D1ACAA64CE509%7D; ASPSESSIONIDQASATDRA=OIOHLMMCCGNJMJPOMONFBFEG; HRLB=172257797751533668</VALUE> </HEADER> </HEADERS> <PARAMETERS> 
<PARAMETER> 
<KEY>UserId</KEY> 
<VALUE>8014371</VALUE> 
</PARAMETER> 
</PARAMETERS> 
</COMPLETEPOSTOPTIONS> 
</INFORMATION> 
============================================================================================================= 
End Contents of Connectivity.EZHex 

Is there anything in your experience that would suggest what the problem is, or what I should be looking at? 

Thanks very much for any pointers. 

Frank 


----- Original Message -----

**Comment by swt2c** [comment]
You don't want to execute the Connectivity.EZHex file directly.  Instead, you want to run it with Congruity.  Does your web browser offer to open the file with Congruity when you download it?  If so, do that.  If not, you can save the file somewhere and then run "congruity <path_to_ezhex_file>" manually.

**Comment by foberle** [comment]
Scott: 

Thanks much for your response. It turns out that Firefox had somehow decided to change the default application for .EZHex files from Congruity to Kate (yeah - rather arbitrary). I'm assuming that since Ubuntu is now a teen-aged operating system it just decided to act as recalcitrant as many teenagers do. 

I renamed ~/.mozilla/firefox/mrxgcr9e.default/mimeTypes.rdf to mimeTypes.rdf.old and restarted Firefox, and that took care of the problem, although I'm a little bit nervous about setting Congruity as the default for .EZHex files again, and will just choose each time for a bit. 

How this occurred is a mystery to me, but since I hadn't used it since trying your new version, I just took a shot at asking you. 

But, once again, thanks very much for your prompt and helpful response. 

Frank 

----- Original Message -----
